static/js/modules/autocomplete.js

```diff
@@ -0,0 +1,115 @@
+import { getAutoCompleteArticles } from "./wikipediaAPI/util.js";
+
+const count = 5;
+
+var autocompleteInput = {
+
+    props: {
+        text: {
+            type: String,
+            default: ""
+        },
+        placeholder: {
+            type: String,
+            default: "Type in an article..."
+        }
+    },
+
+    emits: ['update:text'],
+
+    data: function () {
+        return {
+            acList: [],
+            showAutocomplete: false,
+            highlightIndex: -1
+        }
+	},
+
+    methods: {
+        async setAutocomplete() {
```

Comment:
Another thought, how do you able to deal with multiple in flight requests returning at different times?